commit 7992e00469c485f108a7f0da08be34b0fa441f79
Author: Wenwen Wang <wenwen@cs.uga.edu>
Date:   Mon Aug 19 22:24:50 2019 -0500

    ubifs: Fix memory leak in __ubifs_node_verify_hmac error path
    
    In __ubifs_node_verify_hmac(), 'hmac' is allocated through kmalloc().
    However, it is not deallocated in the following execution if
    ubifs_node_calc_hmac() fails, leading to a memory leak bug. To fix this
    issue, free 'hmac' before returning the error.
    
    Fixes: 49525e5eecca ("ubifs: Add helper functions for authentication support")
    Signed-off-by: Wenwen Wang <wenwen@cs.uga.edu>
    Signed-off-by: Richard Weinberger <richard@nod.at>

diff --git a/fs/ubifs/auth.c b/fs/ubifs/auth.c
index d9af2de9084a..8cdbd53d780c 100644
--- a/fs/ubifs/auth.c
+++ b/fs/ubifs/auth.c
@@ -479,8 +479,10 @@ int __ubifs_node_verify_hmac(const struct ubifs_info *c, const void *node,
 		return -ENOMEM;
 
 	err = ubifs_node_calc_hmac(c, node, len, ofs_hmac, hmac);
-	if (err)
+	if (err) {
+		kfree(hmac);
 		return err;
+	}
 
 	err = crypto_memneq(hmac, node + ofs_hmac, hmac_len);